sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -624,7 +624,11 @@ pi_result _pi_context::finalize() {
     std::scoped_lock<pi_mutex> Lock(EventCacheMutex);
     for (auto &EventCache : EventCaches) {
       for (auto &Event : EventCache) {
-        ZE_CALL(zeEventDestroy, (Event->ZeEvent));
+        auto ZeResult = ZE_CALL_NOCHECK(zeEventDestroy, (Event->ZeEvent));
+        // Gracefully handle the case that L0 was already unloaded.
+        if (ZeResult && ZeResult != ZE_RESULT_ERROR_UNINITIALIZED)
+          return mapError(ZeResult);
```

Comment:
Should we create another macro like ZE_CALL/ZE_CALL_NOCHECK which will not fail in case of ZE_RESULT_ERROR_UNINITIALIZED to use it in finalize/release functions?

Reply:
I don't think so, it's very limited in number of uses and there are cases where you still need to do smth besides destroying L0 handles, e.g. https://github.com/intel/llvm/pull/8948/files#diff-15dd1eb076d2164bd9e87d9057f05f652a716498e8cdf5975e564c65309a0985R7438

Reply:
Ok